### ISSUE TITLE
Failed to upload the Keepass database<database name>

### ISSUE THREAD

**Issue by PA1REG** [KoenZomers/KeePassOneDriveSync#175]
**Describe the bug**
Wanneer ik de database wil synchen dan krijg ik steeds Failed to upload the Keepass database<database name>

**To Reproduce**
Klik op "Save"

**Expected behavior**
Ik vermoed dat de database te groot is geworden. Kan dat?
Verder kan ik je project in c# downloaden maar ik mis referenties zoals "keepass"

**Screenshots**
![2024-12-02 08_13_06-New Issue · KoenZomers_KeePassOneDriveSync en nog 8 andere pagina's - DeSom _ Re](https://github.com/user-attachments/assets/f2f700fb-ec9c-4be1-be0e-6cb54346bb34)

**Versions (please complete the following information):**
Version 2.1.2.2 - November 24, 2021

**Authentication method**
- [x] Microsoft Graph (any browser)
- [ ] Microsoft Graph (built in browser)
- [ ] OneDrive
- [ ] OneDrive for Business
- [ ] SharePoint 2013/2016/2019

**KeePass database synced with**
- [x] OneDrive for Business
- [x] OneDrive Personal

**Comment by beowulf222** [comment]
I experience the same with OneDrive Personal and the built-in browser. Synchronization worked until a week ago when suddenly, I got a "Failed to Upload the KeePass Database" message. The database is about 10MB big.
Interestingly, this only happens with one database. Another one (maybe 1MB) still works.

**Comment by NS8K** [comment]
Hi! I am also experiencing this problem and want to add additional information.
I am running KeePass ver - 2.57.1, DataBaseBackup - ver 2.0.8.6 and KeePass.OneDriveSync - ver 2.1.2.2 on 2 Win 10 desktops and 1 Win 11 laptop.  I only get this error when using the laptop.  Both desktop systems work fine (although of course, they do not receive changes made using the laptop right now).

According to my backup history, on the Win 11 laptop between 12:04 and 14:28 (EST) on 14-Dec-24, the local .kdbx file changed from 87 kb to 17,115 kb. I cannot say the issue started at that time as I just noticed it yesterday (1-Jan-25) but the local .kdbx files on both Win 10 computers remain at 87 kb and they both continue to work.  It is possible I added something large to an entry on the Win 11 laptop at that time and I will search to find if such a thing has happened.

Tom - NS8K

**Comment by NS8K** [comment]
Well, that didn't take long.  There is a convenient search for large entries in KeePass and it quickly pointed out that I had scanned 2 pictures of my wife's and my passports and attached them to our passport entries.  The pic sizes were approximately 4 Mb and after I removed all 4 pictures, I was back to normal operation on the Win 11 laptop.

So, as a test, I added back one 4 Mb .jpg and tried a save.  The error returned.  To complete the test, I removed the attachment a second time and did a save.  THE ERROR DID NOT GO AWAY!  A search for large attachments showed my wife's entry to still be 4 Mb despite the attachment being removed.  The only way I could finally get her entry back to normal size was to re-add both pictures (not just one) and then remove both pictures.  Her entry returned to a small size and synchronization returned to normal operation.

Hope this all helps.  I cannot tell if the issue is the size of the entry or the fact that it was a .jpg attachment or some other issue.  The very strange behavior regarding deleting of an attachment I would guess to be a KeePass defect, not a KeyPass.OneDriveSync issue.

Tom - NS8K

**Comment by jezui** [comment]
> I think the problem is somehow related to the latest KeePass version(s?). I can't say which version exactly, but what I can say is that it worked with version 2.54 and that I updated to 2.57.1 and that causes it to not work anymore.
> 
> And to add to this, I have several PCs and can reproduce this: works with old version - but as soon as I update, it starts giving this error message.

I tried to went back to older version (2.56 - because synchronization worked with) and I got same error. I hope, that the error will be fixed early :)

**Comment by cartesien** [comment]
I searched the error message displayed at the bottom in Koen's code.

I found the message in this page: https://github.com/KoenZomers/KeePassOneDriveSync/blob/master/KoenZomers.KeePass.OneDriveSync/Providers/**OneDriveProvider.cs**

**I think this is an update from Microsoft to their API which connects OnedriveSync to Onedrive in their cloud.**

I'm not a developer and I wouldn't know how to modify the code.

All I can say is that the download of online file is done correctly, the synchronization between the two files (online downloaded and hardrive files) is carried out locally on the machine, but it is the writing of the new file synchronized online which seems not to be carried out. I don't know if it's an authentication or rights problem or modification of the API that Koen uses.

I tried to revert to previous versions which did not solve the problem of sending the file online.

I went back to 2.54 on Windows 11 with the latest Windows updates dating from January 2025. Il have the probleme from 10 january,  when i update keepass to 2.57.1. But it possible that it is a coincidence 🤷‍♂️

Is an AI can help US if Koen doesn't have time to debug and repair?

Translated by google.

**Comment by BlaM** [comment]
> I tried to went back to older version (2.56 - because synchronization worked with) and I got same error. I hope, that the error will be fixed early :)

Interesting, because I'm 100% sure that I had PCs where it failed earlier than others... But the honest answer is: I didn't really test the "version" theory. It was just a feeling.

**Comment by KD12345678** [comment]
Got the same Problem - onedrive sync failed to upload the KeePass Databass. Windows 11 Pro, 24H2 - seems to end Upload since December 2024. Using  2.57.1

**Comment by dyslecix** [comment]
I have the saem problem.:

Windows 10 Pro 19045.5487
Keepass 2.57.1 (64 bit)

Plugins
Keetheme 0.10.7.0
KPEnhancedEntryView 2.6.0.0
SourceForgeUpdateChecker 0.4.0.0
WinHelloUnlock 1.6.1
KoenZomers.Keepass.OneDriveSync 2.1.2.2

Behavior:

When I Save keepass it resolves with the error:

Failed to upload the KeePass database Keepass Database 

(Note: the repitition of 'KeePass database Keepass Database' is intentional as that is exactly the error that shows)

**Comment by PHANT0Maniac** [comment]
The Problem still exist, i think. 
Resetting the OneDriveSync Configuration didn´s help. :-( (Now it´s last synced with = Never synced yet)

The Tool has Problems uploading the Database. Copying the Database manually to the OneDrive-location works.
The Sync (upload) on my smartphone with Keepass2Android works without any problems.

**Comment by jezui** [comment]
I agree, latest version didn't fix the issue. I tried many various attempts but without success. Unfortunately, I'm thinking about other solutions, because I'm using application very often and nothing is happen. :-(

**Comment by KoenZomers** [comment]
Not sure if you guys are still using this plugin, but could you try it again with the latest version I just published? Haven't specifically looked into addressing this issue, but I also cannot reproduce it. I attached a 13 MB file into my KeePass database and it syncs fine using that. If that's not a proper test towards your use case, let me know.

**Comment by rmnwrbl** [comment]
Version 3.1.0.0
<img width="664" height="427" alt="Image" src="https://github.com/user-attachments/assets/6feb1c74-7127-4b64-b11e-eecc9a710e05" />
